gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -41,6 +41,26 @@
 
 #include <pcl/gpu/segmentation/gpu_extract_clusters.h>
 
+#include <cuda_runtime_api.h>
+#include <cuda.h>
+
+pcl::Indices economical_download(const pcl::gpu::NeighborIndices& indices,
+        pcl::Indices& sizes, int max_answers) {
```

Comment:
As commented before, please add a comment for the mut ref, or use a const ref by moving the download outside the function

The order of the arguments could be improved. A possible alternative:
```cpp
namespace pcl {
namespace detail {
// There's no point in performing an allocation. Just out parameter for returning the results
// All these vectors can be further moved out of the big for loop in L96
void economical_download(source_indices, points_per_cluster, result, indices_size, indices_buffer) {
    // do the thing
}
}  // ns details
}  // ns pcl
```

Reply:
I very much agree that the order of the args could be better - thank you! Thank you also for suggesting to pack the function in a namespace. I did all that. Does the function now look better to you?